pyfar/classes/audio.py

```diff
@@ -1296,64 +1300,63 @@ def _arithmetic(data: tuple, domain: str, operation: Callable, **kwargs):
+    """
+    Validates that the type and metadata of input data
+        are appropriate for performing arithmetic operations.
 
-    Input:
+    Parameters
+    ----------
     data : tuple
-        Can contain Signal, TimeData, FrequencyData, and array like data
+        A tuple containing data elements, which can be instances of
+        Signal, TimeData, FrequencyData, or array-like objects.
     domain : str
-        Domain in which the arithmetic operation should be performed. 'time' or
-        'freq'.
+        The domain in which the arithmetic operation is to be performed.
+        Either 'time' or 'freq'.
     division : bool
-        ``True`` if a division is performed, ``False`` otherwise
-    matmul: bool
-        ``True`` if a  matrix multiplication is performed, ``False`` otherwise
+        Indicates if the operation is a division.
+    matmul : bool
+        Indicates if the operation is a matrix multiplication.
 
     Returns
     -------
-    sampling_rate : number, None
-        Sampling rate of the signals. None, if no signal is contained in `data`
-    n_samples : number, None
-        Number of samples of the signals. None, if no signal is contained in
-        `data`
-    fft_norm : str, None
-        FFT norm of the first signal in `data`, if all FFT norms are None.
-        Otherwise the first FFT norm that is not None is taken.
-    times : numpy array, None
-        The times if a TimeData object was passed. None otherwise.
-    frequencies : numpy array, None
-        The frequencies if a FrequencyData object was passed. None otherwise.
-    audio_type : type, None
-        Type of the audio class if contained in data. Otherwise None.
-    cshape : tuple, None
-        Largest channel shape of the audio classes if contained in data.
-        Otherwise empty tuple.
+    tuple
+        A tuple containing validated meta information such as
+        sampling rate, number of samples, FFT norm, etc.
 
+    Raises
+    ------
+    ValueError
+        If the data elements do not conform to expectations
+        based on their type or properties.
+
+    Example
+    --------
+    >>> data = (Signal(...), Signal(...))
+    >>> _assert_match_for_arithmetic(data, 'freq', False, False)
+    (44100, 1024, 'unitary', None, None, Signal, ())
     """
 
     # we need at least two signals
     if not isinstance(data, tuple):
         raise ValueError("Input argument 'data' must be a tuple.")
 
     # check validity of domain
-    if domain not in ['time', 'freq']:
+    if domain not in ["time", "freq"]:
         raise ValueError(f"domain must be time or freq but is {domain}.")
 
     # properties that must match
     sampling_rate = None
     n_samples = None
-    fft_norm = 'none'
+    fft_norm = "none"
     times = None
     frequencies = None
     audio_type = type(None)
     cshape = ()
 
+    # determine operation type based on division flag
+    operation_type = "divide" if division else "multiply"
```

Comment:
This is a bit misleading. The reason we check for the division is that [different rules apply in this case](https://pyfar-gallery.readthedocs.io/en/latest/gallery/interactive/pyfar_arithmetics.html#DFT-normalization-and-arithmetic-operations). But the operation can be addition, subtraction or multiplication. Maybe using the `division` flag as before is cleaner.